From foo@baz Mon Jan  5 12:37:19 PST 2009
Date: Mon, 05 Jan 2009 12:37:19 -0800
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: agnx: fix build errors due to rate control API changes

In commit e6a9854b05c1a6af1308fe2b8c68f35abf28a3ee, the rate control API
was rewritten, so make the needed changes to the agnx driver so that it
will build and work properly.

Cc: Li YanBo <dreamfly281@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/agnx/xmit.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/agnx/xmit.c
+++ b/drivers/staging/agnx/xmit.c
@@ -626,8 +626,8 @@ static inline void fill_agnx_hdr(struct 
 	txhdr->reg0 = cpu_to_be32(reg);
 
 	/* Set the long and short retry limits */
- 	txhdr->tx.short_retry_limit = tx_info->txi->control.retry_limit;
- 	txhdr->tx.long_retry_limit = tx_info->txi->control.retry_limit;
+ 	txhdr->tx.short_retry_limit = tx_info->txi->control.rates[0].count;
+ 	txhdr->tx.long_retry_limit = tx_info->txi->control.rates[0].count;
 
 	/* FIXME */
 	len = tx_info->skb->len - sizeof(*txhdr) + tx_info->hdr_len + FCS_LEN;
@@ -649,7 +649,7 @@ static void txm_power_set(struct agnx_pr
 	u32 reg;
 
 	/* FIXME */
-	if (txi->tx_rate_idx < 0) {
+	if (txi->control.rates[0].idx < 0) {
 		/* For B mode Short Preamble */
 		reg = agnx_set_bits(PHY_MODE, PHY_MODE_SHIFT, AGNX_MODE_80211B_SHORT);
 //		control->tx_rate = -control->tx_rate;
